zerver/tornado/event_queue.py

```diff
@@ -657,20 +658,29 @@ def receiver_is_off_zulip(user_profile_id):
 
 def maybe_enqueue_notifications(user_profile_id, message_id, private_message,
                                 mentioned, stream_push_notify, stream_name,
-                                always_push_notify, idle):
-    # type: (int, int, bool, bool, bool, Optional[str], bool, bool) -> Dict[str, bool]
+                                always_push_notify, idle, group_message):
+    # type: (int, int, bool, bool, bool, Optional[str], bool, bool, bool) -> Dict[str, bool]
     """This function has a complete unit test suite in
     `test_enqueue_notifications` that should be expanded as we add
     more features here."""
     notified = dict()  # type: Dict[str, bool]
 
     if (idle or always_push_notify) and (private_message or mentioned or stream_push_notify):
         notice = build_offline_notification(user_profile_id, message_id)
-        notice['triggers'] = {
-            'private_message': private_message,
-            'mentioned': mentioned,
-            'stream_push_notify': stream_push_notify,
-        }
+        
+        if (group_message and mentioned):
```

Comment:
These strings to cover all the possible cases?

Reply:
Is there a reason to distinct `group_message` from `private_message` here?

Reply:
Basically I wanted to create strings for every possible case, 
If we want to reduce strings we can either go for `notice['mentioned'] = false` and check Recipient.[CASE] and then we don't need triggers.

Reply:
But in the client side, there is no distinction b/w PM or a group message, as the recipient_type has `stream | private` so it could be better to include everything in the trigger field (if no other solution found)